iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs

```diff
@@ -2,43 +2,48 @@
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
 
-use crate::StratagemUpdate;
+use crate::{ActionResponse, StratagemUpdate};
 use bootstrap_components::bs_button;
 use futures::Future;
 use iml_environment::csrf_token;
+use iml_utils::dispatch_custom_event;
 use seed::{class, dom_types::At, fetch, i, prelude::*};
 
 #[derive(Debug, Default)]
 pub struct Model {
     pub config_data: Option<StratagemUpdate>,
+    pub disabled: bool,
 }
 
 #[derive(Clone, Debug)]
 pub enum Msg {
     UpdateStratagem,
-    StratagemUpdated(fetch::FetchObject<iml_wire_types::Command>),
-    OnFetchError(seed::fetch::FailReason<iml_wire_types::Command>),
+    StratagemUpdated(fetch::FetchObject<ActionResponse>),
+    OnFetchError(seed::fetch::FailReason<ActionResponse>),
 }
 
 pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
     match msg {
         Msg::UpdateStratagem => {
             let orders = orders.skip();
 
+            model.disabled = true;
             if let Some(config_data) = &model.config_data {
                 orders.perform_cmd(update_stratagem(&config_data));
             }
         }
         Msg::StratagemUpdated(fetch_object) => match fetch_object.response() {
             Ok(response) => {
                 log::trace!("Response data: {:#?}", response.data);
+                dispatch_custom_event("show_command_modal", &response.data);
                 orders.skip();
             }
             Err(fail_reason) => {
                 orders.send_msg(Msg::OnFetchError(fail_reason)).skip();
             }
         },
         Msg::OnFetchError(fail_reason) => {
+            model.disabled = false;
             log::error!("Fetch error: {:#?}", fail_reason);
             orders.skip();
```

Comment:
You have `orders.skip()` here, which will prevent the button from changing state.

Reply:
removed